experiment/bump_e2e_image.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: bump_e2e_image.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dirty="$(git status --porcelain)"
+if [[ -n ${dirty} ]]; then
+  echo "Tree not clean:"
+  echo "${dirty}"
+  exit 1
+fi
+
+TREE="$(dirname ${BASH_SOURCE[0]})/.."
+
+DATE="$(date +v%Y%m%d)"
+TAG="${DATE}-$(git describe --tags --always --dirty)"
+pushd "${TREE}/jenkins/e2e-image"
+make push TAG="${TAG}"
+popd
+
+sed -i "s/\/kubekins-e2e:.*$/\/kubekins-e2e:${TAG}/" "${TREE}/images/e2e-prow/Dockerfile"
+git commit -am "Bump e2e-prow base image to ${TAG}"
+
+TAG="${DATE}-$(git describe --tags --always --dirty)"
+pushd "${TREE}/images/e2e-prow"
+make push TAG="${TAG}"
+popd
+
+sed -i "s/\/kubekins-e2e-prow:v.*$/\/kubekins-e2e-prow:${TAG}/" "${TREE}/experiment/generate_tests.py"
+pushd "${TREE}"
+bazel run //experiment:generate_tests -- \
+  --yaml-config-path=experiment/test_config.yaml \
+  --json-config-path=jobs/config.json \
+  --prow-config-path=prow/config.yaml
+bazel run //jobs:config_sort
+git commit -am "Re-gen tests with ${TAG}"
+
+# Scan for kubekins-e2e-prow:v.* as a rudimentary way to avoid
+# replacing :latest.
+sed -i "s/\/kubekins-e2e-prow:v.*$/\/kubekins-e2e-prow:${TAG}/" "${TREE}/prow/config.yaml"
+git commit -am "Bump e2e-prow to ${TAG}"
```

Comment:
I can merge them, sure. PArt of that was to differentiate between the test re-gen and the manual bump, but if you think there's no need, it saves a commit.